### ISSUE TITLE
Images created in a different language do not appear in the panel in Home Slider module

### ISSUE THREAD

**Issue by carcam** [PrestaShop/PrestaShop#33527]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I have a 1.7.8.9 PrestaShop using Multishop environment. One shop is meant for people in Spain so it's configured to be in Spanish and the other shop is meant for people in Portugal so it's in Portuguese language. The backoffice management is done in Spanish.

When creating a slider in the Portugal shop context, the slider is saved in portuguese language and it works fine. Nevertheless when I try to see the created slider, it doesn't show in backoffice because the backoffice language is Spanish and no image or data is associated to the slider in Spanish language.

### Expected behavior

I think this doesn't make sense and you need to see all the sliders showing in your shop context no matter the language you are using.

### Steps to reproduce

1. Go to backoffice
2. choose a shop context that uses a different language than the one you see your backoffice in
3. create a new slider and fill only the fields for the shop language
4. save it
5. You will not see the slider in the mangement area.

### PrestaShop version(s) where the bug happened

1.7.8.9

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

ps_homeslider_3.1.3

**Comment by florine2623** [comment]
Hello @carcam ,

You can choose the language here :
<img width="1121" alt="Screenshot 2023-08-02 at 11 23 38" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/2230e473-5ffc-4b24-a237-e4462d8ad120">

And the shop context here :
<img width="375" alt="Screenshot 2023-08-02 at 11 24 18" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/abf31b39-62bc-4e56-9e48-267e1292d831">

I can switch from one language to another without issue.

**Comment by carcam** [comment]
Hi @florine2623,
thank you very much for your reply, but I feel I did not explain the issue correctly. Let me rephrase.

The situation is that when your backoffice user has a different language than the shop default language, you can create the sliders in several languages, following the screenshot you have posted. So far, so good.

But when you create the slider in just the shop context default language and save the slider, that info is stored just for that language and it is not expanded to the other languages of your shop, so, when you check the sliders in the backoffice in a different language than the one you created the slider for, they do not appear as the slider shows only sliders for your shop context that have an image in the language you are using in the backoffice (that doesn't have to be the one you set as default in your shop).

I hope this makes it more clear.

Thanks!!

**Comment by florine2623** [comment]
Yes it is more clear, I forgot a step indeed.

I have dedicated one language per shop :

https://github.com/PrestaShop/PrestaShop/assets/16019289/6031eed4-6da5-4996-82d3-d82ea4ba0fc7

I have edited one slide with French language. My BO language is English. I have modified the French language and checked on the FO. The modifications are well taken into account :

https://github.com/PrestaShop/PrestaShop/assets/16019289/b159cc8a-f544-4810-839d-0be72e1ea95b

Could you send a screen record of your issue as I can't reproduce it ?
Waiting for your feedback.
Thanks!

**Comment by carcam** [comment]
@florine2623 thank you very much for your detailed check into the issue. I see now why you cannot reproduce the issue. When you save your slider, you have filled the fields  in all your available languages. The problem only appears when you fill just one of the languages. Please check this video:

https://github.com/PrestaShop/PrestaShop/assets/1695138/44176a1f-80a1-41bb-947a-2a914fed294e

When we manage our shop for different countries, we usually have the backend in Spanish, and we only need to fill up the fields for the specific language, so it's a bit cumbersome having to upload the same image for all the languages and not being able to see the titles and so on in the backoffice.

**Comment by hibatallahAouadni** [comment]
Hello @carcam 

I reproduce the issue with PS81x and develop, see the attached screen record below:

[issue_33527.webm](https://github.com/PrestaShop/PrestaShop/assets/23308427/42621e0d-9b12-4b4e-a60b-0bb66ed3a0c8)

I’ll add this to the debug backlog so that it’s fixed.

Please be aware that due to the high number of reported issues, some bugs might take a very long time to be resolved. If this issue is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you. 

If you manage to get the problem fixed, then please contribute the fix back to the project as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you

**Comment by Triloworld** [comment]
We have issue as well as on 1.7.8.9 (now on 7.x.x.x)
It has temporary solution: require to add slider language to two languages in BO by user right in edit screen.
To fix issue when happen: Log-in to DB manager and remove slider that are affected and add it once more.

**Comment by carcam** [comment]
This issue is still relevant.

**Comment by boo-code** [comment]
This does not reproduce on PrestaShop 9 any more — @kpodemski's PrestaShop/ps_imageslider#104 fixed it, and I think this issue and PR #106 can both be closed.

Measured on 9.2 with `ps_imageslider` **3.2.2**, creating a slide the way the module itself does and filling only the French fields while the back office context is English:

```
slide saved through Ps_HomeSlide::add()  ->  2 lang rows: lang 1 image=''  lang 2 image='probe.jpg'

BO list, context language en (back office)   4 slides, the new slide visible: YES
BO list, context language fr (shop)          4 slides, the new slide visible: YES
```

The listing query still filters `AND hssl.id_lang = <context language>`, so what makes this work is that both halves hold: `Ps_HomeSlide` is a multilang `ObjectModel`, so saving writes a row for **every** language even when only one is filled in, and #104 added `getSlides(null, true)` for the back office, which drops the `AND hssl.image <> ""` condition that used to hide the row with the empty image. Either one alone would not have been enough — the row has to exist *and* be allowed through without an image.

**A caveat worth recording, because it explains @Triloworld's workaround.** A slide whose lang row is genuinely absent for a language is still not listed, since the query joins on it. That state cannot be produced by saving through the module today, but it is what "remove the affected slider and add it once more" was working around, so shops carrying rows from older versions or hand-edited data may still see it. If anyone can still reproduce this on 9.x, the useful thing to post is `SELECT id_lang FROM ps_homeslider_slides_lang WHERE id_homeslider_slides = <id>` for a slide that is missing from the list — if a language is absent there, it is stale data rather than the current code.

**On PrestaShop/ps_imageslider#106.** @Triloworld's PR adds a validation error when an image is missing for any language, and links this issue. With the listing fixed, that would now make the module stricter than it needs to be — it would force merchants to upload an image in every language for a slide they only want in one, which is the opposite of what @carcam asked for here ("you need to see all the sliders showing in your shop context no matter the language"). It has had no review since 2024-08-09, so it may be worth closing alongside this.

I have not prepared anything, since there is nothing left to fix on the current code.
